articles/actions.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: default
+title: Actions
+permalink: articles/actions.html
+abstract:
+  Actions are one of the three core types of interaction between ROS nodes.
+  This article specifies the requirements for actions, how they've changed from ROS 1, and how they're communicated.
+author: '[Geoffrey Biggs](https://github.com/gbiggs)'
+published: true
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+There are three forms of communication in ROS: topics, services, and actions.
+Topic publishers broadcast to multiple subscribers, but communication is one-way.
+Service clients send a request to a service server and get a response, but there is no information about the progress.
+Similar to services, actions clients ask an action server reach some end state and will get a response.
+Unlike services, while the action is being peformed the action server sends progress updates to the client.
+
+Actions are useful when a response may take a significant length of time.
+They allow a client to track the progress of a request, get the final outcome, and optionally cancel the request it before it completes.
+
+This document defines requirements for actions, how they are specified in the ROS Message IDL, and how they are communicated by the middleware.
+
+## Entities Involved in Actions
+
+There are two entities involved in actions: an action server and an action client.
+
+### Action Server
+
+An action server provides an action.
+Like topics and services, an action server has a name and a type.
+There is only one server per name.
+
+It is responsible for
+
+* advertising the action to other ROS entities
+* accepting or rejecting requests from one or more action clients
```

Comment:
For consistency, we may consider using the term "goal" here (and elsewhere in the doc) when referring to requests from action clients.